pkg/providers/subnet/subnet.go

```diff
@@ -178,9 +180,9 @@ func (p *DefaultProvider) UpdateInflightIPs(createFleetInput *ec2.CreateFleetInp
 	defer p.Unlock()
 
 	// Process the CreateFleetInput to pull out all the requested subnetIDs
-	fleetInputSubnets := lo.Compact(lo.Uniq(lo.FlatMap(createFleetInput.LaunchTemplateConfigs, func(req *ec2.FleetLaunchTemplateConfigRequest, _ int) []string {
-		return lo.Map(req.Overrides, func(override *ec2.FleetLaunchTemplateOverridesRequest, _ int) string {
-			if override == nil {
+	fleetInputSubnets := lo.Compact(lo.Uniq(lo.FlatMap(createFleetInput.LaunchTemplateConfigs, func(req ec2types.FleetLaunchTemplateConfigRequest, _ int) []string {
+		return lo.Map(req.Overrides, func(override ec2types.FleetLaunchTemplateOverridesRequest, _ int) string {
+			if override.SubnetId == nil {
```

Comment:
We control these values. @engedaam Do you know if this is even possible because this check seems weird to me now